homeassistant/components/ozw/light.py

```diff
@@ -118,6 +209,38 @@ async def async_turn_on(self, **kwargs):
         """Turn the device on."""
         self.async_set_duration(**kwargs)
 
+        rgbw = None
+        white = None
+        hs_color = None
+
+        if kwargs.get(ATTR_WHITE_VALUE) is not None:
+            white = kwargs[ATTR_WHITE_VALUE]
+
+        if kwargs.get(ATTR_COLOR_TEMP) is not None:
+            rgbw = "#00000000ff"
+
+        elif kwargs.get(ATTR_HS_COLOR) is not None:
+            hs_color = kwargs[ATTR_HS_COLOR]
+            if ATTR_WHITE_VALUE not in kwargs:
+                # white LED must be off in order for color to work
+                white = 0
+
+        if (
+            kwargs.get(ATTR_WHITE_VALUE) or kwargs.get(ATTR_HS_COLOR)
+        ) and hs_color is not None:
+            rgbw = "#"
+            for colorval in color_util.color_hs_to_RGB(*hs_color):
+                rgbw += format(colorval, "02x")
+            if white is not None:
+                rgbw += format(white, "02x") + "00"
```

Comment:
Please use f-strings as much as possible.

Reply:
I actually tried those at first and was getting an error, I'll give it another shot.